modules/axdma/src/dma.rs

```diff
@@ -0,0 +1,78 @@
+use core::{alloc::Layout, ptr::NonNull};
+
+use allocator::{AllocError, AllocResult, BaseAllocator, ByteAllocator};
+use axalloc::{global_allocator, DefaultByteAllocator};
+use axhal::{mem::virt_to_phys, paging::MappingFlags};
+use kspin::SpinNoIrq;
+use log::{debug, error};
+use memory_addr::{align_up, VirtAddr, PAGE_SIZE_4K};
+
+use crate::{phys_to_bus, DMAInfo};
+
+pub(crate) static ALLOCATOR: SpinNoIrq<DmaAllocator> = SpinNoIrq::new(DmaAllocator::new());
+
+pub(crate) struct DmaAllocator {
+    alloc: DefaultByteAllocator,
+}
+
+impl DmaAllocator {
+    pub const fn new() -> Self {
+        Self {
+            alloc: DefaultByteAllocator::new(),
+        }
+    }
+    /// Allocate arbitrary number of bytes. Returns the left bound of the
+    /// allocated region.
+    ///
+    /// It firstly tries to allocate from the coherent byte allocator. If there is no
+    /// memory, it asks the global page allocator for more memory and adds it to the
+    /// byte allocator.
+    ///
+    /// `align_pow2` must be a power of 2, and the returned region bound will be
```

Comment:
This line can be removed. You can also remove the similar line in `axalloc::GlobalAllocator::alloc`.